packages/synapse-react-client/src/components/ChallengeDetailPage/ChallengeDetailPage.tsx

```diff
@@ -58,13 +77,13 @@ export function ChallengeDetailPage({ projectId }: ChallengeDetailPageProps) {
   }
 
   const handleRequirementsHide = () => {
-    setShowRequirements(false)
-    if (accessToken) {
-      setShowWizard(true)
-    }
+    setShowRegistrationModal(false)
+  }
+  const handleRegistrationComplete = () => {
+    setShowRegistrationModal(false)
+    // Only guide the user to join/create a submission team if they are not already part of one.
+    setShowSubmissionTeamWizard(!isMemberOfSubmissionTeam)
   }
```

Comment:
Before this change, we would show the wizard to join/create a submission team, even if the user was a member of a submission team. In that case, the modal would just show an error, so we can avoid case that entirely.